### PR TITLE
fixing/Node version update

### PR DESCRIPTION
**Pull request by JAYATEJA04**
### **Issue no. #790** 

Now I can pass the integration and unit tests after changing the node-version from 14 to 18 in the package.json and test.yml files. 
### Things I tried to make it work ###

- Changing ports
- Updating the firebase CLI.
- Installing firebase through npm.
- Changing the host address in the firebase.json file. (worked✅)

Before changing the host address in the firebase.json file.
```
{
  "emulators": {
    "firestore": {
      "port": 8081
    },
    "ui": {
      "enabled": true,
      "port": 4000
    }
  }
}
```
### After changing the host address in the firebase.json file ###
```
{
  "emulators": {
    "firestore": {
      "host": "0.0.0.0",
      "port": 8081
    },
    "ui": {
      "host": "0.0.0.0",
      "enabled": true,
      "port": 4000
    }
  }
}
```
Even the node-version has been updated in the test.yml file, and small changes in other files have also been done.

### Reference ### 
1.) https://github.com/Real-Dev-Squad/website-backend/issues/918
2.) https://github.com/firebase/firebase-tools/issues/4741

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -60,13 +60,13 @@
...
   "engines": {
-    "node": "14.x"
+    "node": "18.x"
   },
   "pre-commit": [
     "lint"
   ],
   "volta": {
-    "node": "14.21.1",
+    "node": "18.15.0",
```
Comment (harshith-venkatesh) [inline package.json:69]: Isn't this a major version upgrade, is everything working as expected?
Is this tested thoroughly?

Comment (JAYATEJA04) [reply package.json:69]: Yes, this is a major version upgrade.
As far as testing is considered, I have considered these things:

1. Checking if the backend is successfully set up or not.

image

2. Running unit tests.

image

3. Running integration tests.

image

Are these the tests it needs to pass or is there anything else that is required to be considered? 
Reference: 
https://github.com/Real-Dev-Squad/website-backend/blob/fix-node-version-update/CONTRIBUTING.md#pull-request-guidelines

**Comment by harshith-venkatesh** [comment]
Are all API calls working as expected?

**Comment by JAYATEJA04** [comment]
> > Are all API calls working as expected?
> 
> I'm not aware of how to check API calls, could you please tell me or give me a reference on how to check this one?

@harshith-venkatesh could you please let me know about this?

**Comment by harshith-venkatesh** [comment]
Could that be checked via local database and postman?
@prakashchoudhary07 would you like to review this PR and add more points on the above please

**Comment by JAYATEJA04** [comment]
> Could that be checked via the local database and postman? @prakashchoudhary07 would you like to review this PR and add more points on the above, please

Hello @prakashchoudhary07, could you please let me know about this?

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

**Comment by JAYATEJA04** [comment]
Hello @prakashchoudhary07 
I have made the changes requested, few changes are the same as in the "develop" branch of the website-backend.

**Comment by JAYATEJA04** [comment]
Regarding the change in port in the package.json file, I referred to these two: 
1.) https://github.com/firebase/firebase-tools/issues/4741#issuecomment-1184653807 and 
2.) https://github.com/Real-Dev-Squad/website-backend/issues/918

File: firebase.json [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
   "emulators": {
     "firestore": {
+      "host": "0.0.0.0",
       "port": 8081
     },
     "ui": {
+      "host": "0.0.0.0",
       "enabled": true,
-      "port": 4000
+      "port": 4200
```
Comment (prakashchoudhary07) [inline firebase.json:10]: Why are we changing this PORT

File: package.json [hunk 3]
```diff
@@ -60,13 +60,13 @@
...
     "yarn": "1.22.19"
   }
-}
+}
```
Comment (prakashchoudhary07) [inline package.json:72]: Can you please check why is there a change here

File: firebase.json [hunk 4]
```diff
@@ -1,11 +1,13 @@
 {
   "emulators": {
     "firestore": {
+      "host": "0.0.0.0",
```
Comment (prakashchoudhary07) [inline firebase.json:4]: Why do we need a host here?

File: scripts/tests/tdd-files-list.txt [hunk 5]
```diff
@@ -2,4 +2,4 @@ test/integration/<replace>.test.js
 test/integration/<replace>>.test.js
 test/unit/<replace>/<replace>.test.js
 test/unit/<replace>/<replace>.test.js
-test/unit/<replace>/<replace>.test.js
+test/unit/<replace>/<replace>.test.js
```
Comment (prakashchoudhary07) [inline scripts/tests/tdd-files-list.txt:5]: here as well

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (JAYATEJA04) [reply package.json]: Formatted properly this time and fixed it.

**Review by JAYATEJA04 — COMMENTED** [review]
Edited the necessary changes. Changed the port, fixed the formatting of the files and mentioned the reason behind mentioning host in package.json file.

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**